pkg/setup/setup.go

```diff
@@ -46,60 +46,57 @@ func SetupBacalhauRepo(repoDir string, c config.ReadWriter) (*repo.FsRepo, error
 
 		return nil, fmt.Errorf("failed to check if repo exists: %w", err)
 	} else if !exists {
-		if err := fsRepo.Init(c); err != nil {
+		if err := fsRepo.Init(cfg); err != nil {
 			return nil, fmt.Errorf("failed to initialize repo: %w", err)
 		}
 	} else {
-		if err := fsRepo.Open(c); err != nil {
+		if err := fsRepo.Open(); err != nil {
 			return nil, fmt.Errorf("failed to open repo: %w", err)
 		}
 	}
 	return fsRepo, nil
 }
 
-func SetupBacalhauRepoForTesting(t testing.TB) (*repo.FsRepo, types.BacalhauConfig) {
-	// reset the global viper instance used by cmds pkg.
-	viper.Reset()
-	// create a specific viper instance for testing
-	v := viper.New()
-	// init a config with this viper instance using the local configuration as default
-	cfg, err := config.New(config.WithDefault(configenv.Local))
-	if err != nil {
-		t.Fatal(err)
-	}
-
+func SetupBacalhauRepoForTesting(t testing.TB) (*repo.FsRepo, types.Bacalhau) {
 	// create a temporary dir to serve as bacalhau repo whose name includes the current time to avoid collisions with
 	/// other tests
-	tmpDir, err := os.MkdirTemp("", "")
-	if err != nil {
-		t.Fatal(errors.Wrap(err, "failed to create temporary directory in test setup"))
-	}
+	tmpDir := t.TempDir()
 	path := filepath.Join(tmpDir, fmt.Sprint(time.Now().UnixNano()))
-	// configure the repo on the testing viper insance
-	v.Set("repo", path)
 
-	t.Logf("creating repo for testing at: %s", path)
-	t.Setenv("BACALHAU_ENVIRONMENT", "local")
-	t.Setenv("BACALHAU_DIR", path)
+	// disable update checks in testing.
+	t.Setenv(config.KeyAsEnvVar(types.UpdateConfigIntervalKey), "0")
+	cfgValues := map[string]any{
+		"DataDir": path,
+	}
 
-	// create the repo used for testing
-	fsRepo, err := SetupBacalhauRepo(path, cfg)
+	// the BACALHAU_NODE_IPFS_CONNECT env var is only bound if it's corresponding flags are registered.
+	// This is because viper cannot bind its value to any existing keys via viper.AutomaticEnv() since the
+	// environment variable doesn't map to a key in the config, so we add special handling here until we move away
+	// from this flag to the dedicated flags like "BACALHAU_PUBLISHER_IPFS_ENDPOINT",
+	// "BACALHAU_INPUTSOURCES_IPFS_ENDPOINT", etc which have a direct mapping to the config key based on their name.
+	if connect := os.Getenv("BACALHAU_NODE_IPFS_CONNECT"); connect != "" {
+		cfgValues[types.PublishersIPFSEndpointKey] = connect
+		cfgValues[types.ResultDownloadersIPFSEndpointKey] = connect
+		cfgValues[types.InputSourcesIPFSEndpointKey] = connect
+	}
```

Comment:
Why do we need this? Is this permanent or temporary solution? Isn't `config.New` the right place to set this _if needed_?

Reply:
> Why do we need this?

Please see the code comment above.

>  Is this permanent or temporary solution?

it's required until we stop supporting the environment variable `BACALHAU_NODE_IPFS_CONNECT`.

>  Isn't config.New the right place to set this if needed?

Given your comment elsewhere stating we must maintain behavior of `BACALHAU_NODE_IPFS_CONNECT` I will move this to `config.New`